logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -133,19 +125,20 @@ private Object tryParseBodyAsJson(final String body) {
         return body;
     }
 
-    private boolean isJson(final String contentType) {
-        if (contentType.isEmpty()) {
+    private boolean isJson(final String type) {
+        if (type.isEmpty()) {
             return false;
         }
 
-        final MediaType mediaType = MediaType.parse(contentType);
+        final int pi = type.indexOf(';');
+        final int limit = (pi != -1 ? pi : type.length());
 
-        final boolean isJson = mediaType.is(APPLICATION_JSON);
-
-        final boolean isApplication = mediaType.is(MediaType.ANY_APPLICATION_TYPE);
-        final boolean isCustomJson = mediaType.subtype().endsWith("+json");
+        return type.regionMatches(true, 0, "application/json", 0, limit)
```

Comment:
I haven't tried it, but I think the custom sub type should be non-greedy
On 14 May 2016 11:50, "Jörn Horstmann" <notifications@github.com> wrote:

> In
> logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java
> <https://github.com/zalando/logbook/pull/88#discussion_r63273802>:
>
> >
> > -        final boolean isJson = mediaType.is(APPLICATION_JSON);
> > -
> > -        final boolean isApplication = mediaType.is(MediaType.ANY_APPLICATION_TYPE);
> > -        final boolean isCustomJson = mediaType.subtype().endsWith("+json");
> > +        return type.regionMatches(true, 0, "application/json", 0, limit)
>
> Might be more readable as a regular expression like
>
> final Pattern MEDIA_TYPE_JSON_PATTERN = Pattern.compile("application/(?:json|.+\\+json)(?:$|\\s|;)");
>
> return MEDIA_TYPE_JSON_PATTERN.matcher(type).lookingAt();
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/zalando/logbook/pull/88/files/96c8d2457061745659dfc7cd8d6a5af77a888a67#r63273802>
>